src/scenarios/shared/runner.py

```diff
@@ -691,7 +692,37 @@ def run(self):
                     f'--launchdevbundleid={self.packagename}',
                 ]
                 runCmdCommand = RunCommand(runCmd, verbose=True)
-                runCmdCommand.run()
+
+                try:
+                    runCmdCommand.run()
+                except CalledProcessError as ex:
+                    if ex.returncode == 70:
+                        # Exit code 70 from xharness means time out, this can happen sometimes when the device is in a screwed state
+                        # and doesn't correctly launch apps anymore. In that case we reboot the device
+                        getLogger().error("Device is in a broken state, rebooting.")
+                        rebootCmd = xharnesscommand() + [
+                            'apple',
+                            'mlaunch',
+                            '--',
+                            '--rebootdev',
+                        ]
+                        rebootCmdCommand = RunCommand(rebootCmd, verbose=True)
+                        rebootCmdCommand.run()
+
+                        getLogger().info("Waiting 30 secs for the device to boot.")
+                        time.sleep(30)
```

Comment:
Is there any way to check if the reboot has completed? if so it might be better to change this to a loop that checks every 5 seconds if the reboot is complete, just in case it is faster than 30 seconds.

Reply:
There's no good way unfortunately. Given this happens quite rarely I don't think it's an issue if we waste a few seconds when the phone reboots more quickly.